jwst/resample/resample_spec.py

```diff
@@ -91,73 +92,155 @@ def build_interpolated_output_wcs(self, refmodel=None):
-        warnings.resetwarnings()
-
-        spectral_axis = find_dispersion_axis(refmodel)
-        spatial_axis = spectral_axis ^ 1
-
-        # Compute the wavelength array, trimming NaNs from the ends
-        wavelength_array = np.nanmedian(lam, axis=spectral_axis)
-        wavelength_array = wavelength_array[~np.isnan(wavelength_array)]
-
-        # Compute RA and Dec up the slit (spatial direction) at the center
-        # of the dispersion.  Use spectral_axis to determine slicing dimension
-        lam_center_index = int((bb[spectral_axis][1] - bb[spectral_axis][0]) / 2)
-        if not spectral_axis:
-            x_tan_array = x_tan.T[lam_center_index]
-            y_tan_array = y_tan.T[lam_center_index]
-        else:
-            x_tan_array = x_tan[lam_center_index]
-            y_tan_array = y_tan[lam_center_index]
-        x_tan_array = x_tan_array[~np.isnan(x_tan_array)]
-        y_tan_array = y_tan_array[~np.isnan(y_tan_array)]
-
-        fitter = LinearLSQFitter()
-        fit_model = Linear1D()
-        pix_to_ra = fitter(fit_model, np.arange(x_tan_array.shape[0]), x_tan_array)
-        pix_to_dec = fitter(fit_model, np.arange(y_tan_array.shape[0]), y_tan_array)
-
+        # for each input model convert slit x,y to ra,dec,lam
+        # use first input model to set spatial scale
+        # use center of appended ra and dec arrays to set up
+        # center of final ra,dec
+        # append all ra,dec, wavelength array for each slit
+        # use first model to initialize wavelenth array
+        # append wavelengths that fall outside the endpoint of
+        # of wavelength array when looping over additional data
+
+        all_wavelength = []
+        all_ra_slit = []
+        all_dec_slit = []
+
+        for im, model in enumerate(self.input_models):
+            wcs = model.meta.wcs
+            bb = wcs.bounding_box
+            grid = wcstools.grid_from_bounding_box(bb)
+            ra, dec, lam = np.array(wcs(*grid))
+            spectral_axis = find_dispersion_axis(model)
+            spatial_axis = spectral_axis ^ 1
+
+            # Compute the wavelength array, trimming NaNs from the ends
+            wavelength_array = np.nanmedian(lam, axis=spectral_axis)
+            wavelength_array = wavelength_array[~np.isnan(wavelength_array)]
+
+            # need to estimate the spatial sampling to use for the output WCS
+            # it is assumed the spatial sampling is the same for all the input
+            # models. So we can use the first input model to set the spatial
+            # sampling.
+            # Note: using the first input model sets up the spatial output to match
+            # this  model. For nodded (background subtracted data)
+            # this means the positive and negative traces will have the same
+            # configurations as the first input model.
+
+            # Steps to do this for first input model:
+            # 1. find the center of slit in wavelength
```

Comment:
I know these are just comments and hence don't affect the actual processing, but when I read "center of slit", I immediately picture in my mind a physical, rectangular slit, that has its long axis in the spatial direction (which is natural for a spectrograph, because you want to he spectral dimension of the slit to be narrow) and hence the center of the slit refers to something in the spatial (long) dimension of the slit itself. I think what you're referring to here is the center/middle of the spectral trace along the wavelength axis, right? So perhaps this should be reworded along those lines, e.g. find the middle of the spectrum in wavelength. It's the spectrum we're operating on, not the entrance slit.